File: analyze.py

from termcolor import colored as c

def are_files_equal(file1, file2):
    # Open both files in binary mode
    with open(file1, 'rb') as f1, open(file2, 'rb') as f2:
        # Read and compare the content byte by byte
        while True:
            byte1 = f1.read(1)
            byte2 = f2.read(1)
            if byte1 != byte2:
                return False
            if not byte1:
                break  # Reached the end of both files

    return True

# Usage example
file1 = "/Users/michaeltotaro/LivestreamDirectory/database/og_song_list.json"
file2 = "/Users/michaeltotaro/LivestreamDirectory/database/song_list.json"

if are_files_equal(file1, file2):
    print("SONG_LIST EQUAL: YES")
else:
    print(c("The files are not equivalent.", 'red'))

file1Content = []

with open(file1, 'r') as file1r:
    for line in file1r:
        file1Content.append(line)

file2Content = []
with open(file2, 'r') as file2r:
    for line in file2r: 
        file2Content.append(line)

for i in range(len(file1Content)):
    if file1Content[i] != file2Content[i]:
        print()
        print()
        print(f"1. {file1Content[i]}")
        print(f"2. {file2Content[i]}")

file1 = "/Users/michaeltotaro/LivestreamDirectory/db_manager/json_files/artists.json"
file2 = "/Users/michaeltotaro/LivestreamDirectory/temp/artists.json"

if are_files_equal(file1, file2):
    print("artists.json EQUAL: YES")
else:
    print(c("The files are not equivalent.", 'red'))


file1 = "/Users/michaeltotaro/LivestreamDirectory/db_manager/json_files/no_repeats.json"
file2 = "/Users/michaeltotaro/LivestreamDirectory/temp/no_repeats.json"

if are_files_equal(file1, file2):
    print("no_repeats.json EQUAL: YES")
else:
    print(c("The files are not equivalent.", 'red'))


file1 = "/Users/michaeltotaro/LivestreamDirectory/db_manager/json_files/only_with_keys.json"
file2 = "/Users/michaeltotaro/LivestreamDirectory/temp/only_with_keys.json"

if are_files_equal(file1, file2):
    print("only_with_keys EQUAL: YES")
else:
    print(c("The files are not equivalent.", 'red'))

file1 = "/Users/michaeltotaro/LivestreamDirectory/db_manager/json_files/repertoire.json"
file2 = "/Users/michaeltotaro/LivestreamDirectory/temp/repertoire.json"

if are_files_equal(file1, file2):
    print("only_with_keys EQUAL: YES")
else:
    print(c("The files are not equivalent.", 'red'))

file1Content = []

with open(file1, 'r') as file1r:
    for line in file1r:
        file1Content.append(line)

file2Content = []
with open(file2, 'r') as file2r:
    for line in file2r: 
        file2Content.append(line)

for i in range(len(file1Content)):
    if file1Content[i] != file2Content[i]:
        print(f"1. {file1Content[i]}")
        print(f"2. {file2Content[i]}")

File: analyze.rb
# Adding self makes the method a static Class Method.
def c(txt, color) 

    #Program gave threw an error when I tried declaring these as constants
    red = "\e[31m"
    blue = "\e[34m"
    magenta = "\e[35m"
    cyan = "\e[36m"
    reset = "\e[0m"

    case color
        when "red"
            return "#{red}#{txt}#{reset}"
        when "blue"
            return "#{blue}#{txt}#{reset}"
        when "magenta"
            return "#{magenta}#{txt}#{reset}"
        when "cyan"
            return "#{cyan}#{txt}#{reset}"
        else
            print c("ERROR: ", "red")
            puts "Invalid input"
            exit(0)
    end

end

def are_files_equal(file1_content, file2_content)
    return file1_content == file2_content
end 

# Usage example
file1 = "../LivestreamDirectory/database/og_song_list.json"
file2 = "../LivestreamDirectory/database/song_list.json"

if are_files_equal(File.read(file1), File.read(file2))
    puts "SONG_LIST EQUAL: YES")
else
    puts c("The files are not equivalent.", 'red'))
end

file1_content = []
file2_content = []

File.foreach(file1) do |line|
    file1_content.push(line)
end 

File.foreach(file2) do |line|
    file2_content.push(line)
end 

for i in 0...file1_content.length
    if file1_content[i] != file2_content[i]
        puts "\n\n\n"
        puts "1. #{file1_content[i]}"
        puts "2. #{file2_content[i]}"
    end 
end 

file1 = "../LivestreamDirectory/db_manager/json_files/artists.json"
file2 = "../LivestreamDirectory/temp/artists.json"


if are_files_equal(File.read(file1), File.read(file2))
    puts "artists.json EQUAL: YES"
else
    puts c("The files are not equivalent.", 'red')
end 

file1 = "../LivestreamDirectory/db_manager/json_files/no_repeats.json"
file2 = "../LivestreamDirectory/temp/no_repeats.json"

if are_files_equal(File.read(file1), File.read(file2))
    puts "no_repeats.json EQUAL: YES")
else
    puts c("The files are not equivalent.", 'red'))
end

file1 = "../LivestreamDirectory/db_manager/json_files/only_with_keys.json"
file2 = "../LivestreamDirectory/temp/only_with_keys.json"

if are_files_equal(File.read(file1), File.read(file2))
    puts "only_with_keys EQUAL: YES"
else
    puts c("The files are not equivalent.", 'red')
end 

file1 = "../LivestreamDirectory/db_manager/json_files/repertoire.json"
file2 = "../LivestreamDirectory/temp/repertoire.json"

if are_files_equal(File.read(file1), File.read(file2))
    puts "only_with_keys EQUAL: YES"
else
    puts c("The files are not equivalent.", 'red'))
end 

file1_content = []
file2_content = []
        
File.foreach(file1) do |line|
    file1_content.push(line)
end 

File.foreach(file2) do |line|
    file2_content.push(line)
end 

for i in 0...file1_content.length
    if file1_content[i] != file2_content[i]
        print("1. #{file1Content[i]}")
        print("2. #{file2Content[i]}")
    end 
end 